Gems/PhysX/Code/Source/EditorHeightfieldColliderComponent.h

```diff
@@ -69,6 +88,22 @@ namespace PhysX
         AZStd::shared_ptr<Physics::HeightfieldShapeConfiguration> m_shapeConfig{ aznew Physics::HeightfieldShapeConfiguration() };
         //! Contains all of the runtime logic for creating / updating / destroying the heightfield collider.
         AZStd::unique_ptr<HeightfieldCollider> m_heightfieldCollider;
+
+        bool m_useBakedHeightfield = false;
+        AZStd::string m_bakedHeightfieldRelativePath;
+        AZ::Data::Asset<Pipeline::HeightFieldAsset> m_bakedHeightfieldAsset;
+        AZStd::recursive_mutex m_bakedHeightfieldMutex;
+
+        class HeightfieldBakingJob : public AZ::Job
```

Comment:
question: Would it be better to use the new task system here or is AZ:Job preferred as this is a longer running task?

Reply:
These involve file I/O, so best suited for the Job system rather than tasks, but good call to keep in mind we have alternatives and in some cases task graph system is the right choice!

Reply:
Got you thanks!